lib/content_negotiation_filter.js

```diff
@@ -0,0 +1,100 @@
+"use strict";
+
+const HyperSwitch = require('hyperswitch');
+const semver = require('semver');
+const HTTPError = HyperSwitch.HTTPError;
+const URI = HyperSwitch.URI;
+const mwUtil = require('./mwUtil');
+
+module.exports = (hyper, req, next, options, specInfo) => {
+    const rp = req.params;
+    if (!mwUtil.isHTMLRoute(specInfo.path)) {
+        // The filter is  only supported on HTML routes,
+        // but don't punish the client for our own mistake, just log an error.
+        hyper.logger.log('error/configuration',
+            `Content negotiation filter is not supported on ${specInfo.path}`);
+        return next(hyper, req);
+    }
+
+    if (mwUtil.isNoCacheRequest(req)) {
+        // Ignore for the no-cache requests.
+        return next(hyper, req);
+    }
+
+    let requestedVersion = mwUtil.extractHTMLProfileVersion(req.headers.accept);
+    if  (!requestedVersion) {
+        // 1. Lack of accept header
+        // 2. Malformed accept header
+        // 3. False-positive is possible in case of
+        // content-type mismatch, not just version.
+        return next(hyper, req);
+    }
+    // We ignore the patch version, so if it's specified, replace it with 0
+    requestedVersion = requestedVersion.replace(/\d+$/, '0');
+    return next(hyper, req)
+    .then((htmlRes) => {
+        const storedVersion =
+            mwUtil.extractHTMLProfileVersion(htmlRes.headers['content-type']);
+        if (!storedVersion) {
+            // TODO: This can happen if sections are requested;
+            // Think what to do with that.
+            // If it's not the sections - something is very wrong on our side.
+            return htmlRes;
+        }
+
+        if (semver.satisfies(storedVersion, requestedVersion)) {
```

Comment:
This might be clearer as `eq` since the `requestedVersion` isn't a range.

But maybe you meant to use a partial or x-range (to ignore the patch)?

Reply:
We ignore the patch both in Varnish (by basically resetting the patch to 0) and in RESTBase

Reply:
https://github.com/wikimedia/restbase/pull/1069